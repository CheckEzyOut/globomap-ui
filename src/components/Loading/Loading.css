/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.ui-loading {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.3);
  z-index: 99;
}

.ui-loading .ui-loading-cog {
  color: rgba(255, 255, 255, 1)
}

.ui-loading-enter {
  opacity: 0.01;
}

.ui-loading-enter.ui-loading-enter-active {
  opacity: 1;
  transition: opacity 200ms ease-in;
}

.ui-loading-exit {
  opacity: 1;
}

.ui-loading-exit.ui-loading-exit-active {
  opacity: 0.01;
  transition: opacity 200ms ease-in;
}
