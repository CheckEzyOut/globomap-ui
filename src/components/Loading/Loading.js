/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import './Loading.css';

export class Loading extends React.Component {
  render() {
    const iconSizes = { 'small': 'fa-1x', 'medium': 'fa-2x','big': 'fa-3x' };
    const iconSize = Object.keys(iconSizes).includes(this.props.iconSize)
                      ? iconSizes[this.props.iconSize]
                      : iconSizes['medium'];
    return (
      this.props.isLoading &&
        <div className="ui-loading">
          <i className={`ui-loading-cog fa fa-cog fa-spin ${iconSize} fa-fw`}></i>
        </div>
    );
  }
}

export default Loading;
