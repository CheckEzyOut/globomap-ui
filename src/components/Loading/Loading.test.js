import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from './Loading';

describe("Loading", function() {

  it('should render Loading component', () => {
    const component = shallow(<Loading />);
    expect(component).toMatchSnapshot();
  });

});
