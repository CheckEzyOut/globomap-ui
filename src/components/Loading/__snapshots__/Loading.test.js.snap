// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render Loading component 1`] = `
<CSSTransitionGroup
  transitionAppear={false}
  transitionEnter={true}
  transitionEnterTimeout={200}
  transitionLeave={true}
  transitionLeaveTimeout={200}
  transitionName="ui-loading"
/>
`;
